pkg/operator/ceph/nfs/nfs.go

```diff
@@ -219,27 +222,39 @@ func (r *ReconcileCephNFS) createConfigMap(n *cephv1.CephNFS, name string) (stri
 
 // Down scale the ganesha server
 func (r *ReconcileCephNFS) downCephNFS(n *cephv1.CephNFS, nfsServerListNum int) error {
-	diffCount := nfsServerListNum - n.Spec.Server.Active
-	for i := 0; i < diffCount; {
-		depIDToRemove := nfsServerListNum - 1
+	for i := nfsServerListNum - 1; i >= n.Spec.Server.Active; i-- {
+		idToRemove := i
 
```

Comment:
Reworked this loop to run backward from high to low IDs. There was no reason for the loop to be so complicated as to calculate `diffCount` and then count upwards from there. Or to track `i++` separately from the loop definition.